services/autoscaling/src/simcore_service_autoscaling/utils_docker.py

```diff
@@ -2,163 +2,179 @@
+            }
         )
-    if "NanoCPUs" in reservations:
-        cpu = int(reservations["NanoCPUs"]) / 1_000_000_000
-    return {"RAM": ram, "CPU": cpu}
 
+    return Resources.parse_obj(dict(cluster_resources_counter))
 
-async def check_tasks_resources(nodes_ids: list[str]) -> TasksResources:
-    total_tasks_cpus = 0
-    total_tasks_ram = 0
-    tasks_resources = []
-    total_pending_tasks_cpus = 0
-    total_pending_tasks_ram = 0
-    tasks_pending_resources = []
 
-    async with aiodocker.Docker() as docker:
+def get_resources_from_docker_task(task: Task) -> Resources:
+    assert task.Spec  # nosec
+    if task.Spec.Resources and task.Spec.Resources.Reservations:
+        return Resources(
+            cpus=(task.Spec.Resources.Reservations.NanoCPUs or 0) / _NANO_CPU,
+            ram=parse_obj_as(
+                ByteSize, task.Spec.Resources.Reservations.MemoryBytes or 0
+            ),
+        )
+    return Resources(cpus=0, ram=ByteSize(0))
+
 
-        services = await docker.services.list()
-        for service in services:
-            tasks = service.tasks()
-            for task in tasks:
-                if (
-                    task["Status"]["State"] == "running"
-                    and task["NodeID"] in nodes_ids
-                    and (
-                        reservations := task["Spec"]
-                        .get("Resources", {})
-                        .get("Reservations")
-                    )
-                ):
-                    tasks_resources.append(
-                        {"ID": task["ID"], **_get_reserved_resources(reservations)}
-                    )
-
-                elif (
-                    task["Status"]["State"] == "pending"
-                    and task["Status"]["Message"] == "pending task scheduling"
-                    and "insufficient resources on" in task["Status"]["Err"]
-                    and (
-                        reservations := task["Spec"]
-                        .get("Resources", {})
-                        .get("Reservations")
-                    )
-                ):
-                    tasks_pending_resources.append(
-                        {"ID": task["ID"], **_get_reserved_resources(reservations)}
-                    )
-
-        total_tasks_cpus = 0
-        total_tasks_ram = 0
-        for task in tasks_resources:
-            total_tasks_cpus += task["CPU"]
-            total_tasks_ram += task["RAM"]
-
-        for task in tasks_pending_resources:
-            total_pending_tasks_cpus += task["CPU"]
-            total_pending_tasks_ram += task["RAM"]
-
-        return TasksResources(
-            total_cpus_running_tasks=total_tasks_cpus,
-            total_ram_running_tasks=total_tasks_ram,
-            total_cpus_pending_tasks=total_pending_tasks_cpus,
-            total_ram_pending_tasks=total_pending_tasks_ram,
-            count_tasks_pending=len(tasks_pending_resources),
+async def compute_node_used_resources(node: Node) -> Resources:
```

Comment:
MINOR: you are not "computing" anything , just listing resources. I would stick to the CRUD vocabulary on the names (i.e. create, list, get, update, delete  ...) Here it would be ``list_node_used_resources``


Same applies for the rest of function names. IMO all API calls should (if possible) be standardize into this CRUD verbs

Reply:
well you are wrong here. I do compute (i.e. add each service reservations).. AFAIK that is still a computation no?